docs/reference/projectyml.rst

```diff
@@ -100,72 +113,111 @@ example of a dev config:
 .. code-block:: yaml
 
     pip:
-        index-url: https://devpi.example.com/dev/
+        index-url: https://my_repo.secure.company.com/repository/dev
         extra-index-url: []
         pre: true
         use-system-config: false
 
 2) In production
 
-Using a single pip index is still the recommended approach, and the use of pre-release versions should be disabled. Here is an
-example of a config suitable in production:
+Using a single pip index is still the recommended approach, and the use of pre-release versions should be disabled.
+
+
+If the pip configuration is fully managed at the system level and secure, use ``use-system-config: true``
+and add any potential project-specific repo that is not managed at the system level through the ``extra-index-url``
+option e.g.:
 
 .. code-block:: yaml
 
     pip:
-        index-url: https://devpi.example.com/stable/
-        extra-index-url: []
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-production]
         pre: false
         use-system-config: true
 
-.. note::
-    The options defined in the ``project.yml`` pip section will always take precedence over the corresponding pip options, even
-    when ``use-system-config`` is set to true (other pip-related environment variables are not overridden).
-
-    For example, in the production scenario above, if the following
-    pip environment variables were set by mistake on the server running the compiler: ``PIP_INDEX_URL=https://devpi.example.com/dev/``
-    and ``PIP_PRE=true``, the config used in the end would still be the one defined in the project.yml, namely
-    ``index-url=https://devpi.example.com/stable/`` and ``pre=false``.
 
-An alternative approach would be to configure all pip-related options through the system config. For example:
+On the other hand, if the pip configuration is not fully managed at the system level, make sure to disable ``use-system-config``
```

Comment:
I think this is still our envisioned approach / recommendation. So I would move this up and rephrase (because now it's relative to the parapraph before it). For the `use-system-config=true` section then I would not advocate it per se, but phrase it more like "if you prefer to manage ...".